### PR TITLE
cpython: drop python 3.5/3.6 logic, patches, packages

### PR DESCRIPTION
**Pull request by mweinelt**
###### Description of changes

Make things nice and clean again.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
Can we drop `python3Packages.easysnmp`?

**Comment by mweinelt** [comment]
Probably also a good time to drop `functools32`, `enum34`, `backports_ssl_match_hostname`, maybe more.

**Comment by mweinelt** [comment]
This is all for now.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1096,8 +1096,6 @@ self: super: with self; {
 
   backports-cached-property = callPackage ../development/python-modules/backports-cached-property { };
 
-  backports_abc = callPackage ../development/python-modules/backports_abc { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:1099]: Missing alias entry

File: pkgs/development/python-modules/sniffio/default.nix [hunk 2]
```diff
@@ -1,27 +1,22 @@
...
 
   buildInputs = [ glibcLocales ];
 
-  propagatedBuildInputs = lib.optionals (pythonOlder "3.7") [ contextvars ];
-
-  checkInputs = [ pytest curio ];
-
-  checkPhase = ''
-    pytest
-  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sniffio/default.nix:24]: In this commit are a few unrelated changes which could maybe be moved to their own commits.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/top-level/python-packages.nix:1099]: Pointless, since they only work on removed python versions.
